### PR TITLE
Better receiver Main code

### PR DESCRIPTION
**Pull request by slinkydeveloper**
## Proposed Changes

- :broom: Cleanup receiver `Main`, added some comments
- :broom: Moved counter creation to `ReceiverVerticleFactory`

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
/retest

Webhook flaky

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/Main.java [hunk 1]
```diff
@@ -76,9 +66,9 @@
    * @param args command line arguments.
    */
   public static void main(final String[] args) throws IOException {
-    final var env = new ReceiverEnv(System::getenv);
+    ReceiverEnv env = new ReceiverEnv(System::getenv);
 
-    final OpenTelemetrySdk openTelemetry = TracingConfig.fromDir(env.getConfigTracingPath()).setup();
```
Comment (matzew) [inline data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/Main.java:81]: I actually like `final` to state this is final ...

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold
(for release cut and others to review)

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/Main.java (hunk 1 above)
Comment (slinkydeveloper) [reply data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/Main.java:81]: But from Java 8+, this is already implicit when used in closures. Also, in this context, this is bootstrapping code, so obviously everything created here is final and not overwritten :)

Comment (matzew) [reply data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/Main.java:81]: true - more a nit and pers. pref :-)

**Review by pierDipi — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**